save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/TestSuitesController.kt

```diff
@@ -49,15 +51,72 @@ class TestSuitesController(
     /**
      * @param testSuiteIds
      * @param authentication
+     * @param isContest
      * @return [Flux] of [TestSuiteDto]s
      */
     @PostMapping("/api/$v1/test-suites/get-by-ids")
     fun getTestSuitesByIds(
         @RequestBody testSuiteIds: List<Long>,
+        @RequestParam(required = false, defaultValue = "false") isContest: Boolean,
         authentication: Authentication,
     ): Flux<TestSuiteDto> = testSuitesService.findTestSuitesByIds(testSuiteIds)
+        .mapToDtoFiltered(authentication, isContest)
+
+    /**
+     * @param organizationName
+     * @param authentication
+     * @param isContest
+     * @return [Flux] of [TestSuiteDto]s
+     */
+    @GetMapping("/api/$v1/test-suites/get-by-organization")
+    fun getTestSuitesByOrganizationName(
+        @RequestParam organizationName: String,
+        @RequestParam(required = false, defaultValue = "false") isContest: Boolean,
+        authentication: Authentication,
+    ): Flux<TestSuiteDto> = testSuitesService.findTestSuitesByOrganizationName(organizationName)
+        .mapToDtoFiltered(authentication, isContest)
+
+    /**
+     * @param authentication
+     * @param isContest
+     * @return [Flux] of [TestSuiteDto]s
+     */
+    @GetMapping("/api/$v1/test-suites/get-standard")
+    fun getStandardTestSuites(
+        @RequestParam(required = false, defaultValue = "false") isContest: Boolean,
+        authentication: Authentication,
+    ): Mono<TestSuiteDtoList> = testSuitesService.getStandardTestSuites()
+        .map { testSuites ->
+            testSuites.filter {
+                if (isContest) {
+                    it.plugins == PluginType.contestAllowedPlugins
+                } else {
+                    it.plugins.isNotEmpty()
+                }
+            }
+        }
+
+    /**
+     * @param authentication
+     * @param isContest
+     * @return [Flux] of [TestSuiteDto]s
+     */
+    @GetMapping("/api/$v1/test-suites/available")
+    fun getPublicTestSuites(
+        @RequestParam(required = false, defaultValue = "false") isContest: Boolean,
+        authentication: Authentication,
+    ): Flux<TestSuiteDto> = testSuitesService.findAllTestSuites()
+        .mapToDtoFiltered(authentication, isContest)
+
+    private fun Flux<TestSuite>.mapToDtoFiltered(authentication: Authentication, isContest: Boolean): Flux<TestSuiteDto> = filter { testSuite ->
+        testSuitePermissionEvaluator.canAccessTestSuite(testSuite, authentication)
+    }
         .filter { testSuite ->
-            testSuitePermissionEvaluator.canAccessTestSuite(testSuite, authentication)
+            if (isContest) {
+                testSuite.pluginsAsListOfPluginType() == PluginType.contestAllowedPlugins
```

Comment:
It can, but the point is that there are two places where we have almost the same lines of code. The only difference is that here we deal with `TestSuite`, but in other place it's `TestSuiteDto`

Reply:
I mean, it would be nice to have a method that checks if test suite can belong to a contest, instead of manually comparing plugin types right at the usage site

Reply:
Oh, will add it

Reply:
Done